amid/amos/dataset.py

```diff
@@ -0,0 +1,122 @@
+from functools import lru_cache
+from pathlib import Path
+from zipfile import ZipFile
+
+import nibabel
+import numpy as np
+import pandas as pd
+from connectome import Source, meta
+
+from ..internals import checksum, register
+from ..utils import open_nii_gz_file, unpack
+from .utils import add_labels
+
+
+ARCHIVE_NAME = 'amos22.zip'
+ARCHIVE_ROOT_NAME = 'amos22'
+
+
+@register(
+    body_region=('Abdomen'),
+    license='CC BY 4.0',
+    link='https://zenodo.org/record/7262581',
+    modality=('CT', 'MRI'),
+    raw_data_size='23G',
+    prep_data_size='23G',
+    task='Supervised multi-modality abdominal multi-organ segmentation',
+)
+@checksum('amos')
+class AMOS(Source):
+    """
+    AMOS provides 500 CT and 100 MRI scans collected from multi-center, multi-vendor, multi-modality, multi-phase,
+    multi-disease patients, each with voxel-level annotations of 15 abdominal organs, providing challenging examples
+    and test-bed for studying robust segmentation algorithms under diverse targets and scenarios. [1]
+
+    Parameters
+    ----------
+    root : str, Path, optional
+        Absolute path to the root containing the downloaded archive and meta.
+        If not provided, the cache is assumed to be already populated.
+
+    Notes
+    -----
+    Download link: https://zenodo.org/record/7262581/files/amos22.zip
+
+    Examples
+    --------
+    >>> # Download the archive and meta to any folder and pass the path to the constructor:
+    >>> ds = AMOS(root='/path/to/the/downloaded/files')
+    >>> print(len(ds.ids))
+    # 961
+    >>> print(ds.image(ds.ids[0]).shape)
+    # (768, 768, 90)
+    >>> print(ds.mask(ds.ids[26]).shape)
+    # (512, 512, 124)
+
+    References
+    ----------
+    .. [1] JI YUANFENG. (2022). Amos: A large-scale abdominal multi-organ benchmark for
+    versatile medical image segmentation [Data set]. Zenodo. https://doi.org/10.5281/zenodo.7262581
+    """
+
+    _root: str = None
+
+    add_labels(locals())
+
+    @meta
+    def ids(_id2split):
+        return sorted(_id2split)
+
+    def image(i, _id2split, _root):
+        file = f'images{_id2split[i]}/amos_{i}.nii.gz'
+
+        with unpack(Path(_root) / ARCHIVE_NAME, file, ARCHIVE_ROOT_NAME) as (unpacked, is_unpacked):
```

Comment:
it would be better if the root pointed directly to the archive, because it can be renamed by the user